2007-02-07  Kristian Rietveld  <kris@imendio.com>
    
    	* gtk/gtkfilechooserutils.[ch]
    	(hildon_gtk_file_chooser_install_properties): conditionalize with
    	MAEMO_CHANGES, port over doc comment.

2006-12-04  Tommi Komulainen  <tommi.komulainen@nokia.com>
    
    	Export enough symbols for implementing GtkFileChooser interface
    	outside gtk.
    
    	* gtk/gtk.symbols: add hildon_gtk_file_chooser_install_properties
    	* gtk/Makefile.am (INCLUDES): add -DGTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
    	(gtk_private_h_sources, gtk_semi_private_h_sources): export
    	gtkfilesystemmodel.h, gtkfilechooserprivate.h, and
    	gtkfilechooserutils.h as semi-private
    	* gtk/gtkfilechooserutils.c
    	(hildon_gtk_file_chooser_install_properties): add exported version of
    	_gtk_file_chooser_install_properties
    	* gtk/gtkfilesystemmodel.h: add GTK_FILE_SYSTEM_ENABLE_UNSUPPORTED
    	#ifdef guard
    	* gtk/gtkfilechooserprivate.h: add GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
    	#ifdef guard
    	* gtk/gtkfilechooserutils.h: add GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
    	#ifdef guard, declare hildon_gtk_file_chooser_install_properties
    
    	* tests/autotestfilechooser.c: build with
    	GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
Index: gtk+2.0-2.11.6/gtk/Makefile.am
===================================================================
--- gtk+2.0-2.11.6.orig/gtk/Makefile.am	2007-07-20 22:12:29.000000000 +0200
+++ gtk+2.0-2.11.6/gtk/Makefile.am	2007-07-23 13:02:14.000000000 +0200
@@ -33,6 +33,8 @@
 	-I$(top_srcdir)/gdk-pixbuf -I../gdk-pixbuf	\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
+	-DMAEMO_CHANGES					\
+	-DGTK_FILE_CHOOSER_ENABLE_UNSUPPORTED		\
 	-DGTK_FILE_SYSTEM_ENABLE_UNSUPPORTED		\
 	-DGTK_PRINT_BACKEND_ENABLE_UNSUPPORTED		\
 	$(GTK_DEBUG_FLAGS)				\
@@ -335,7 +337,10 @@
 # that are not included in gtk/gtk.h
 gtk_semi_private_h_sources =    \
 	gtktextlayout.h		\
-	gtkfilesystem.h
+	gtkfilesystem.h		\
+	gtkfilesystemmodel.h	\
+	gtkfilechooserprivate.h	\
+	gtkfilechooserutils.h
 
 # GTK+ header files that don't get installed
 gtk_private_h_sources =		\
@@ -350,10 +355,7 @@
 	gtkfilechooserdefault.h	\
 	gtkfilechooserembed.h	\
 	gtkfilechooserentry.h	\
-	gtkfilechooserprivate.h	\
 	gtkfilechoosersettings.h \
-	gtkfilechooserutils.h	\
-	gtkfilesystemmodel.h	\
 	gtkfilesystemunix.h	\
 	gtkhsv.h		\
 	gtkiconcache.h		\
Index: gtk+2.0-2.11.6/gtk/gtk.symbols
===================================================================
--- gtk+2.0-2.11.6.orig/gtk/gtk.symbols	2007-07-20 22:12:29.000000000 +0200
+++ gtk+2.0-2.11.6/gtk/gtk.symbols	2007-07-23 13:02:14.000000000 +0200
@@ -1416,6 +1416,12 @@
 #endif
 #endif
 
+#if IN_HEADER(__GTK_FILE_CHOOSER_UTILS_H__)
+#if IN_FILE(__GTK_FILE_CHOOSER_UTILS_C__)
+hildon_gtk_file_chooser_install_properties
+#endif
+#endif
+
 #if IN_HEADER(__GTK_FILE_CHOOSER_WIDGET_H__)
 #if IN_FILE(__GTK_FILE_CHOOSER_WIDGET_C__)
 gtk_file_chooser_widget_get_type G_GNUC_CONST
Index: gtk+2.0-2.11.6/gtk/gtkfilechooserprivate.h
===================================================================
--- gtk+2.0-2.11.6.orig/gtk/gtkfilechooserprivate.h	2007-07-20 22:12:28.000000000 +0200
+++ gtk+2.0-2.11.6/gtk/gtkfilechooserprivate.h	2007-07-23 13:02:14.000000000 +0200
@@ -21,6 +21,14 @@
 #ifndef __GTK_FILE_CHOOSER_PRIVATE_H__
 #define __GTK_FILE_CHOOSER_PRIVATE_H__
 
+/* This is a "semi-private" header; it is meant only for
+ * alternate GtkFileChooser implementations; no stability guarantees 
+ * are made at this point
+ */
+#ifndef GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
+#error "gtkfilechooserprivate.h is not supported API for general use"
+#endif
+
 #include "gtkfilechooser.h"
 #include "gtkfilesystem.h"
 #include "gtkfilesystemmodel.h"
Index: gtk+2.0-2.11.6/gtk/gtkfilechooserutils.c
===================================================================
--- gtk+2.0-2.11.6.orig/gtk/gtkfilechooserutils.c	2007-07-20 22:12:29.000000000 +0200
+++ gtk+2.0-2.11.6/gtk/gtkfilechooserutils.c	2007-07-23 13:02:14.000000000 +0200
@@ -119,6 +119,22 @@
 				    "do-overwrite-confirmation");
 }
 
+#ifdef MAEMO_CHANGES
+/**
+ * hildon_gtk_file_chooser_install_properties:
+ *
+ * Exactly the same as the private _gtk_file_chooser_install_properties()
+ * but exported for hildon-fm.
+ *
+ * Since: maemo 2.0
+ */
+void
+hildon_gtk_file_chooser_install_properties (GObjectClass *klass)
+{
+  _gtk_file_chooser_install_properties (klass);
+}
+#endif /* MAEMO_CHANGES */
+
 /**
  * _gtk_file_chooser_delegate_iface_init:
  * @iface: a #GtkFileChoserIface structure
@@ -360,3 +376,6 @@
   g_signal_emit_by_name (data, "confirm-overwrite", &conf);
   return conf;
 }
+
+#define __GTK_FILE_CHOOSER_UTILS_C__
+#include "gtkaliasdef.c"
Index: gtk+2.0-2.11.6/gtk/gtkfilechooserutils.h
===================================================================
--- gtk+2.0-2.11.6.orig/gtk/gtkfilechooserutils.h	2007-07-20 22:12:29.000000000 +0200
+++ gtk+2.0-2.11.6/gtk/gtkfilechooserutils.h	2007-07-23 13:02:14.000000000 +0200
@@ -22,6 +22,14 @@
 #ifndef __GTK_FILE_CHOOSER_UTILS_H__
 #define __GTK_FILE_CHOOSER_UTILS_H__
 
+/* This is a "semi-private" header; it is meant only for
+ * alternate GtkFileChooser implementations; no stability guarantees 
+ * are made at this point
+ */
+#ifndef GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
+#error "gtkfilechooserutils.h is not supported API for general use"
+#endif
+
 #include "gtkfilechooserprivate.h"
 
 G_BEGIN_DECLS
@@ -45,6 +53,9 @@
 } GtkFileChooserProp;
 
 void _gtk_file_chooser_install_properties (GObjectClass *klass);
+#ifdef MAEMO_CHANGES
+void hildon_gtk_file_chooser_install_properties (GObjectClass *klass);
+#endif /* MAEMO_CHANGES */
 
 void _gtk_file_chooser_delegate_iface_init (GtkFileChooserIface *iface);
 void _gtk_file_chooser_set_delegate        (GtkFileChooser *receiver,
Index: gtk+2.0-2.11.6/gtk/gtkfilesystemmodel.h
===================================================================
--- gtk+2.0-2.11.6.orig/gtk/gtkfilesystemmodel.h	2007-07-20 22:12:28.000000000 +0200
+++ gtk+2.0-2.11.6/gtk/gtkfilesystemmodel.h	2007-07-23 13:02:14.000000000 +0200
@@ -21,6 +21,10 @@
 #ifndef __GTK_FILE_SYSTEM_MODEL_H__
 #define __GTK_FILE_SYSTEM_MODEL_H__
 
+#ifndef GTK_FILE_SYSTEM_ENABLE_UNSUPPORTED
+#error "GtkFileSystemModel is not supported API for general use"
+#endif
+
 #include <glib-object.h>
 #include "gtkfilesystem.h"
 #include <gtk/gtktreemodel.h>
Index: gtk+2.0-2.11.6/tests/autotestfilechooser.c
===================================================================
--- gtk+2.0-2.11.6.orig/tests/autotestfilechooser.c	2007-07-20 22:13:57.000000000 +0200
+++ gtk+2.0-2.11.6/tests/autotestfilechooser.c	2007-07-23 13:02:32.000000000 +0200
@@ -30,6 +30,7 @@
  */
 
 #define GTK_FILE_SYSTEM_ENABLE_UNSUPPORTED
+#define GTK_FILE_CHOOSER_ENABLE_UNSUPPORTED
 #undef GTK_DISABLE_DEPRECATED
 
 #include <config.h>
